presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -116,16 +120,18 @@ public InternalResourceGroupManager(
             ServerConfig serverConfig)
     {
         requireNonNull(queryManagerConfig, "queryManagerConfig is null");
+        this.queryManagerConfig = requireNonNull(queryManagerConfig, "queryManagerConfig is null");
         this.exporter = requireNonNull(exporter, "exporter is null");
         this.configurationManagerContext = new ResourceGroupConfigurationManagerContextInstance(memoryPoolManager, nodeInfo.getEnvironment());
-        this.legacyManager = requireNonNull(legacyManager, "legacyManager is null");
-        this.configurationManager = new AtomicReference<>(cast(legacyManager));
+        this.initializingConfigurationManager = new InitializingConfigurationManager();
+        this.configurationManager = new AtomicReference(initializingConfigurationManager);
```

Comment:
nit: there is a generics warning on usage of raw types.
Should we use something similar to previous code
`this.configurationManager = new AtomicReference<>(cast(initializingConfigurationManager));`